app/src/main/java/com/gathering/android/event/home/EventListViewModel.kt

```diff
@@ -13,23 +14,36 @@ class EventListViewModel @Inject constructor() : ViewModel() {
     val viewState: LiveData<EventViewState> by ::_viewState
 
     fun onViewCreated() {
+        loadEventList()
+    }
+
+    fun onFilterChanged(filter: Filter) {
+        loadEventList(filter)
+    }
+
+    private fun loadEventList(filter: Filter = Filter()) {
         _viewState.setValue(EventViewState.ShowProgress)
-        val list = provideEventList()
-        if (list.isNotEmpty()) {
-            _viewState.setValue(EventViewState.ShowEventList(list))
-            _viewState.setValue(EventViewState.HideNoData)
-        } else {
+        val filteredEventList = provideEventList(
+            filter
+        )
+
+        _viewState.setValue(EventViewState.ShowEventList(filteredEventList))
+        if (filteredEventList.isEmpty()) {
             _viewState.setValue(EventViewState.ShowNoData)
+        } else {
+            _viewState.setValue(EventViewState.HideNoData)
         }
-        _viewState.setValue(EventViewState.HideProgress)
 
+        _viewState.setValue(EventViewState.HideProgress)
     }
 
     fun onEventItemClicked(event: Event) {
         EventViewState.NavigateToEventDetail(event)
     }
 
-    private fun provideEventList(): List<Event> {
+    private fun provideEventList(
+        filter: Filter
+    ): List<Event> {
         val eventList = mutableListOf<Event>()
```

Comment:
lets move the "providing a mock Lis"t business to another method and rename this method to something like: `getEventList` or `getRefinedEventList`